criu/cr-restore.c

```diff
@@ -2213,7 +2213,7 @@ static int restore_root_task(struct pstree_item *init)
 		goto out_kill_network_unlocked;
 	}
 
-	if (lazy_pages_finish_restore())
+	if (!opts.no_background_fetch && lazy_pages_finish_restore())
```

Comment:
I'm not sure I understand how this would prevent background fetch

Reply:
Yes, I found I can not prevent background fetch.  Even I apply the following patch.
```
diff --git a/criu/uffd.c b/criu/uffd.c
index 98c2b7e07..0ca4cdbaa 100644
--- a/criu/uffd.c
+++ b/criu/uffd.c
@@ -68,6 +68,7 @@
 #define MAX_XFER_LEN    (4 << 20)
 
 static mutex_t *lazy_sock_mutex;
+static bool disable_background_lazy_page_fetch;
 
 struct lazy_iov {
        struct list_head l;
@@ -1391,11 +1392,15 @@ static int prepare_uffds(int listen, int epollfd)
                        goto close_uffd;
        }
 
-       lazy_sk_rfd.fd = client;
-       lazy_sk_rfd.read_event = lazy_sk_read_event;
-       lazy_sk_rfd.hangup_event = lazy_sk_hangup_event;
-       if (epoll_add_rfd(epollfd, &lazy_sk_rfd))
-               goto close_uffd;
+    if (!disable_background_lazy_page_fetch) {
+           lazy_sk_rfd.fd = client;
+           lazy_sk_rfd.read_event = lazy_sk_read_event;
+           lazy_sk_rfd.hangup_event = lazy_sk_hangup_event;
+           if (epoll_add_rfd(epollfd, &lazy_sk_rfd))
+                   goto close_uffd;
+    } else {
+           restore_finished = true;
+    }
 
        close(listen);
        return 0;
@@ -1475,3 +1480,8 @@ int cr_lazy_pages(bool daemon)
        xfree(events);
        return ret;
 }
+
+static void __attribute__((constructor)) background_lazy_page_fetch_init(void)
+{
+    disable_background_lazy_page_fetch = (getenv("CRIU_DISABLE_BACKGROUND_LAZY_PAGE_FETCH") != NULL);
+}

```